hledger-lib/Hledger/Data/Types.hs

```diff
@@ -208,13 +213,22 @@ instance Show AmountStyle where
     (show asdecimalpoint)
     (show asdigitgroups)
 
+data AmountPrecision = Precision !Word8 | NaturalPrecision deriving (Eq,Ord,Read,Show,Typeable,Data,Generic)
+
+instance NFData AmountPrecision
+
```

Comment:
Semigroup instance for AmountPrecision: sums the two precisions, capping the result at the maximum of 255. I see